source/dem/velocity_verlet_integrator.cc

```diff
@@ -130,5 +130,113 @@ VelocityVerletIntegrator<dim>::integrate(
     }
 }
 
+template <int dim>
+void
+VelocityVerletIntegrator<dim>::integrate(
+  Particles::ParticleHandler<dim> &particle_handler,
+  const Tensor<1, 3> &             g,
+  const double                     dt,
+  std::vector<Tensor<1, 3>> &      torque,
+  std::vector<Tensor<1, 3>> &      force,
+  const std::vector<double> &      MOI,
+  const typename DEM::dem_data_structures<dim>::cell_set
+    &mobile_cells_container)
+{
+  Point<3>           particle_position;
+  const Tensor<1, 3> dt_g = g * dt;
+
+  for (auto &particle : particle_handler)
+    {
+      // Get the total array view to the particle properties once to improve
+      // efficiency
+      types::particle_index particle_id         = particle.get_local_index();
+      auto                  particle_properties = particle.get_properties();
+
+      // Check if the particle is mobile, if not, integration is skipped
+      auto &cell_particle = particle.get_surrounding_cell();
```

Comment:
This is an expensive call (getting the surrounding cell)
It might be better to write this integrate with the cell set instead by looping over the cells and then looping over the particles within these cells.